tests/common/devices/sonic.py

```diff
@@ -642,31 +637,47 @@ def all_critical_process_status(self):
             service_group_process = group_process_results[service]
 
             service_result = service_results[service]
-            # If container is not running, stdout_lines is empty
+
+            all_critical_process[service] = self.get_service_status_and_critical_process(
+                service_result=service_result,
+                service_critical_process=service_critical_process,
+                critical_group_list=service_group_process['groups'],
+                critical_process_list=service_group_process['processes']
+            )
+
+        return all_critical_process
+
+    def get_service_status_and_critical_process(self, service_result, service_critical_process, critical_group_list,
+                                     critical_process_list):
+        """
+        Parse the result of command "docker exec <container_name> supervisorctl status"
+        and get service container status and critical processes
+        """
+        # If container is not running, stdout_lines is empty
+        # In this situation, service container status should be false
+        if not service_result['stdout_lines']:
+            service_critical_process['status'] = False
```

Comment:
Try not to return value by updating the passed in dict in place. This is implicit and error prone.
Suggest to initialize a new dict. Parsed passed in output content. Update the dict, and eventually return this dict.